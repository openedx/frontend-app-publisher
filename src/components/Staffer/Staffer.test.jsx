import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import { Staffer } from './index';

const defaultProps = {
  onRemove: jest.fn(),
  item: {
    uuid: '2aba6189-ad7e-45a8-b269-bea071b80391',
    given_name: 'Dave',
    family_name: 'Grohl',
    profile_image_url: '/media/people/profile_images/2aba6189-ad7e-45a8-b269-bea071b80391-11df6812f839.png',
  },
  referrer: '/course/11111111-1111-1111-1111-111111111111',
};

describe('Staffer', () => {
  it('renders the staffer', () => {
    const component = shallow(<Staffer {...defaultProps} />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('calls onRemove when the delete button is clicked', () => {
    const component = shallow(<Staffer {...defaultProps} />);
    component.find('.js-delete-btn').simulate('click');
    expect(defaultProps.onRemove).toHaveBeenCalled();
  });
});
