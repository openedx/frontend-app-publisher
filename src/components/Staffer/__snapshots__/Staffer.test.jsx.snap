// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Staffer renders the staffer 1`] = `
<Fragment>
  <div
    className="staffer-image-wrapper overflow-hidden"
  >
    <img
      alt=""
      className="rounded-circle w-25"
      src="/media/people/profile_images/2aba6189-ad7e-45a8-b269-bea071b80391-11df6812f839.png"
    />
  </div>
  <div
    className="staffer-details"
  >
    <button
      className="btn js-delete-btn mr-1 p-0"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <SvgDelete
        className="text-gray-500"
        id="delete-icon-2aba6189-ad7e-45a8-b269-bea071b80391"
        screenReaderText="Remove Dave Grohl"
      />
    </button>
    <Link
      className="btn mr-1 p-0"
      onClick={[Function]}
      to="/instructors/2aba6189-ad7e-45a8-b269-bea071b80391"
    >
      <SvgEdit
        className="text-gray-500"
        id="edit-icon-2aba6189-ad7e-45a8-b269-bea071b80391"
        screenReaderText="Edit Dave Grohl"
        title="Edit"
      />
    </Link>
    <span
      className="name font-weight-bold"
    >
      Dave Grohl
    </span>
  </div>
</Fragment>
`;
